src/webhook/regular_user_validation.py

```diff
@@ -25,22 +31,33 @@ def handle_request():
     return responses.response_invalid()
   
   try:
+    body_dict = request.json['request']
+    username = body_dict['userInfo']['username']
+
+    if body_dict['object'] is None:
+      group_name = body_dict['oldObject']['metadata']['name']
+    else:
+      group_name = body_dict['object']['metadata']['name']
+
     # get the username and decide if it's a special user (SA, kube:admin, etc) or not.. 
     # deny any request if it's not a special user:  kube:*, system:* except system:unauthenticated
+    # ALLOW any request if the group_name is in admin_groups
     # reference: https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#request
-    body_dict = request.json['request']
-    username = body_dict['userInfo']['username']
-    if username == "system:unauthenticated" or (not username.startswith("kube:") and not username.startswith("system:")):
+    if is_request_allowed(username, group_name, admin_groups):
       kind = body_dict['object']['kind']
       operation = body_dict['operation']
-      namespace = None
-      if 'namespace' in body_dict:
-        namespace = body_dict['namespace']
       return responses.response_deny(req=body_dict, msg="Regular user '{}' cannot {} kind '{}'.".format(username, operation, kind))
     else:
       return responses.response_allow(req=body_dict)
   except Exception:
     print("Exception:")
     print("-"*60)
     traceback.print_exc(file=sys.stdout)
-    return responses.response_invalid()
\ No newline at end of file
+    return responses.response_invalid()
+
+
+def is_request_allowed(username, groupname, admin_groupnames=[]):
+  if groupname not in admin_groupnames and (username == "system:unauthenticated" or (not username.startswith("kube:") and not username.startswith("system:"))):
+    return False
+  else:
+    return True
```

Comment:
Probably could.  My initial try didn't work so I just did this to get it done..

Reply:
IMHO it would be more scrutable if you split it up _further_.
```
if groupname in admin_groupnames:
    return True
if username == "system:authenticated":
    return True
if username.split(":", 1)[0] in ("kube", "system"):
    return True
return False
```

Reply:
Pushed an update to `return not(<the if condition>)`.
Suggestions welcome, but I think it's reasonable to be able to read it as it is.